hunts/src/tagFilter.js

```diff
@@ -0,0 +1,24 @@
+function getDescendants(tableRow) {
+  const allChildren = [tableRow];
+  for (let i = 0; i < allChildren.length; i++) {
+    const current = allChildren[i];
+    Array.prototype.push.apply(allChildren, current.subRows);
+  }
+  return allChildren;
+}
+
+export function addOrRemoveTag(tagList, tag) {
+  let tagInList = tagList.filter(x => ['name', 'color', 'id'].every(key => x[key] === tag[key]))[0];
+  return (tagInList !== undefined) ? tagList.filter((i) => i !== tagInList) : tagList.concat(tag);
+}
+
+export function filterPuzzlesByTagfn(rows, id, filterValue) {
```

Comment:
nit: Should we change `Tagfn` to `TagFn` for upper camel case? The existing `filterTypes` don't seem to be consistent, but uppercase `Fn` seems slightly better?